### PR TITLE
HAAR-470: Download endpoint

### PR DESCRIPTION
**Pull request by ravmoj**
Add report based (for CSV) of list of u…sers based on same criteria as used by BETA user search

### CODE REVIEW

File: src/test/resources/application-test.yml [hunk 1]
```diff
@@ -13,6 +13,9 @@ api:
 spring:
   flyway:
     enabled: true
+    properties:
```
Comment (Mjwillis) [inline src/test/resources/application-test.yml:16]: Remove?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/UserPersonDetail.kt [hunk 2]
```diff
@@ -195,6 +236,20 @@ fun UserPersonDetail.toUserSummaryWithEmail() = UserSummaryWithEmail(
...
+fun UserPersonDetail.toDownloadUserSummaryWithEmail() = UserSummaryWithEmail(
+  username = username,
+  staffId = staff.staffId,
+  firstName = staff.firstName.capitalizeFully(),
+  lastName = staff.lastName.capitalizeFully(),
+  active = isActive(),
+  status = accountDetail?.status,
+  locked = accountDetail?.isLocked() ?: false,
+  expired = accountDetail?.isExpired() ?: false,
+  activeCaseload = null,
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/UserPersonDetail.kt:248]: Still need the caseload in the report

Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/UserPersonDetail.kt:248]: Will need to map twice the column with update false

**Review by Mjwillis — CHANGES_REQUESTED** [review]
need to return the 3 digital caseload

File: src/test/resources/application-test.yml (hunk 1 above)
Comment (ravmoj) [reply src/test/resources/application-test.yml]: removed

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/UserPersonDetail.kt (hunk 2 above)
Comment (ravmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/UserPersonDetail.kt]: Added case load id. Can you please elaborate on last comment ('update')?

**Review by Mjwillis — APPROVED** [review]
LGTM

**Pull request merged**